src/posts/posts.controller.spec.ts

```diff
@@ -0,0 +1,18 @@
+import { Test, TestingModule } from '@nestjs/testing';
+import { PostsController } from './posts.controller';
+
+describe('PostsController', () => {
+  let controller: PostsController;
+
+  beforeEach(async () => {
+    const module: TestingModule = await Test.createTestingModule({
+      controllers: [PostsController],
+    }).compile();
+
+    controller = module.get<PostsController>(PostsController);
+  });
+
+  it('should be defined', () => {
+    expect(controller).toBeDefined();
+  });
+});
```

Comment:
The following tests should be added to test the posts controller sufficiently:
- Test if all endpoints are implemented
```
 it('should have createPost, getAllPosts, createOptionGroup, flagPost & deletePost methods', () => {
    expect(controller).toHaveProperty('createPost');
    expect(controller).toHaveProperty('getAllPosts');
    expect(controller).toHaveProperty('createOptionGroup');
    expect(controller).toHaveProperty('flagPost');
    expect(controller).toHaveProperty('deletePost');
  });
  ```
- Test the functionality of createpost

```
 describe('CreatePost', () => {
    it('Should throw not implemented error', () => {
      expect(controller.createPost).toThrowError(new NotImplementedException());
    });
  });
```